packages/backend/src/models/GroupsModel.ts

```diff
@@ -106,58 +103,63 @@ export class GroupsModel {
-                }
+        const group = await this.getGroup(groupUuid);
 
-                return memberQuery.select(
-                    'group_memberships.group_uuid',
-                    'users.user_uuid',
-                    'users.first_name',
-                    'users.last_name',
-                    'emails.email',
-                );
-            })
+        const membersQuery = this.database
+            .from(GroupMembershipTableName)
             .innerJoin(
-                'organizations',
-                'groups.organization_id',
-                'organizations.organization_id',
+                UserTableName,
+                `${GroupMembershipTableName}.user_id`,
+                `${UserTableName}.user_id`,
             )
-            .leftJoin('members', 'groups.group_uuid', 'members.group_uuid')
-            .where('groups.group_uuid', groupUuid)
-            .select();
+            .innerJoin(
+                EmailTableName,
+                `${UserTableName}.user_id`,
+                `${EmailTableName}.user_id`,
+            )
+            .innerJoin(
+                OrganizationTableName,
+                `${GroupMembershipTableName}.organization_id`,
+                `${OrganizationTableName}.organization_id`,
+            )
+            .where(`${GroupMembershipTableName}.group_uuid`, groupUuid)
+            .andWhere(`${EmailTableName}.is_primary`, true);
 
-        if (rows.length === 0) {
-            throw new NotFoundError(`No group found`);
+        const memberProfilesQuery = membersQuery.clone();
+
+        if (includeMembers !== undefined) {
+            memberProfilesQuery.limit(includeMembers);
         }
+
+        if (offset !== undefined) {
+            memberProfilesQuery.offset(offset);
+        }
+
+        const memberProfiles = await memberProfilesQuery.select(
+            `${UserTableName}.user_uuid`,
+            `${UserTableName}.first_name`,
+            `${UserTableName}.last_name`,
+            `${EmailTableName}.email`,
+        );
+
+        const memberUuids = await membersQuery.select(
+            this.database.raw(
+                `ARRAY_AGG(${UserTableName}.user_uuid) as member_uuids`,
+            ),
+        );
+
         return {
-            uuid: rows[0].group_uuid,
-            name: rows[0].name,
-            createdAt: rows[0].created_at,
-            organizationUuid: rows[0].organization_uuid,
-            members: (rows[0].user_uuid ? rows : []).map((row) => ({
+            uuid: group.uuid,
+            name: group.name,
+            createdAt: group.createdAt,
+            organizationUuid: group.organizationUuid,
+            members: memberProfiles.map((row) => ({
                 userUuid: row.user_uuid,
                 email: row.email,
                 firstName: row.first_name,
                 lastName: row.last_name,
             })),
+            memberUuids: memberUuids?.[0]?.member_uuids ?? [],
```

Comment:
What would be the difference between this and the FE using the members array? `group.members.map(member => member.userUuid)`

Reply:
@ZeRego,
`members` primarily displays the user name and avatar information and has a limit. It includes a query argument, for example, includeMembers=5, which selects only 5 members.

The "memberUuids" serves two purposes:
1. Displaying the total number of users within the group in the UI.
2. Deriving group-inherited permissions, as shown here for orgs: ![CleanShot 2024-01-15 at 22 00 41@2x](https://github.com/lightdash/lightdash/assets/962095/d6f361fb-3a6d-4b1c-b53a-1112cb807e4d)

Reply:
Got it 👍 wasn't understanding the impact of  `includeMembers=5` on this. 
As a developer who consumes the `GroupWithMembers` type, I have no clue that `members` might be incomplete. Perhaps we need to rename that prop to `previewMembers` or have a different a new type for this scenario.
This is something we need to fix in a separate PR.

Reply:
that's a good point. we'll have to work further on the API once we do more UI work around groups. I'll have a separate PR for it